feat: Add migration guide for new language handling (#1930)
shopware/docs: guides/plugins/plugins/translations/built-in-translation-system.md

> # Built-in Translation Handling

Reviewer: Most other articles use Start Case for h1 case-style, so I'll leave this uppercase

> If you are currently using the LanguagePack plugin, please refer to the [Migration Guide](/) for instructions on switching to the new system.

## Where do the translations come from?

Translations are fetched from a public GitHub repository: [shopware/translations](https://github.com/shopware/translations).
This repository is managed using [Crowdin](https://crowdin.com/project/shopware6) and contains translations for Shopware as well as for some official plugins.
The repository syncs with Crowdin every day to ensure that the latest translations are always available.

## How to Install and Update Translations?

To use the built-in translation system, you can utilize the following console commands:

### Install Translations

The `translation:install` command is used to download and install translations for Shopware and its plugins from the configured GitHub repository.
It allows you to specify which locales to install, whether to install all available locales, and whether to skip activation of the installed translations.
Re-installing an already installed locale will override its translations.

```bash
$ php bin/console translation:install [--all, --locales, --skip-activation]

> # example

Reviewer: Why prefix the bash comment with multiple comment signs? For highlighting?

> # example

Reviewer 2: That's not a comment sign, that's a headline in markdown and since the headline above ist h3, this has to be h4

$ php bin/console translation:install --locales=fr-FR,pl-PL --skip-activation
```

### Update Command

The `translation:update` command is used to update existing translations for Shopware and its plugins from the configured GitHub repository.

```bash
$ php bin/console translation:update
```

## Language Activation

By default, installed translations are automatically activated, making them available for use in Shopware.
If you want to install translations without activating them, you can use the `--skip-activation` option with the `translation:install` command.
The `active` flag in the `language` table indicates whether a language is active or not. If a language is not active, it will not be available for selection in the administration or storefront.

## How does the system recognize new updates?

The translations repository includes a [metadata file](https://github.com/shopware/translations/blob/main/crowdin-metadata.json) which contains metadata information about the translations, such as locale, last update timestamps, and percentage of completion for each language.
The `updatedAt` field in the metadata is used to determine if a translation needs to be updated.

> Installing or updating translations will create or update a `crowdin-metadata.lock` file on your private filesystem which stores the last known update timestamps for each locale.

Reviewer 3: There were some discussions about read-only filesystem ([1](https://github.com/shopware/shopware/pull/12497), [https://github.com/shopware/shopware/issues/12398](2)) and I'm wondering if we need to explicitly mention this or maybe event update the implementation itself?

Reviewer 4: @bojanrajh you might want to update the link, weird thing happening with it ->

<img width="1482" height="392" alt="image" src="https://github.com/user-attachments/assets/be343b2c-d632-4bfe-bd35-a7ff3a96db34" />

Reviewer: @marcelbrode @bojanrajh You insist on a note, although the lock file is written where the invoices are written and if that directory is not writable, the Shop has problem more urgent than translations?

---

#### `languages`

**Type:** `array[object]`  
**Example:**

```yaml
languages:
  - name: 'Français'
    locale: 'fr-FR'
```

**Fields:**

- `name` (`string`): Human-readable language name (preferably in the native script).
- `locale` (`string`): Language code according to [IETF BCP 47](https://datatracker.ietf.org/doc/html/bcp47), used by Shopware for translations.

**Description:**  
Defines the set of supported languages for which translations should be retrieved. Each entry specifies a display name and a locale code.

---

## How to Extend or Modify the Configuration Handling

### TranslationConfigLoader

The `TranslationConfigLoader` (`src/Core/System/Snippet/Service/TranslationConfigLoader.php`) is part of the Shopware core and is responsible for loading and validating the `translation.yaml` file. It provides a `TranslationConfig` entity that contains all configured fields from the `translation.yaml`.
It ensures that URLs are valid, languages and plugins are properly structured, and plugin mappings are resolved. Errors such as missing files or invalid configuration values are raised as `SnippetException`.

> To extend or modify its behavior, the decoration pattern is used: services should depend on the abstract class `AbstractTranslationConfigLoader`, and custom decorators can override methods like `load()` or the configuration path while delegating to the original loader.

Reviewer 2: ```suggestion
To extend or modify its behavior, the Decorator Pattern is used: services should depend on the abstract class `AbstractTranslationConfigLoader`, and custom decorators can override methods like `load()` or the configuration path while delegating to the original loader.
```

### TranslationConfig

The `TranslationConfig` (`src/Core/System/Snippet/Service/TranslationConfig.php`) is a data structure that holds the configuration details loaded from the `translation.yaml` file.

> You can require it via dependency injection and because of the usage of the `TranslationConfigLoader` with lazy loading, the configuration is always available when needed.

Reviewer 2: ```suggestion
You can require it via dependency injection, and thanks to the TranslationConfigLoader with lazy loading, the configuration is always available when needed.
```
